### ISSUE TITLE
str.splitlines() does not strip \r in \r\n line endings

### ISSUE THREAD

**Issue by mbuesch** [micropython/micropython#1689]
CPython 3.4:
```
>>> 'abc\r\n'.splitlines()
['abc']
```

MicroPython git
```
>>> 'abc\r\n'.splitlines()
['abc\r']
```

**Comment by dpgeorge** [comment]
Just to provide a reply on this.  Implementing "full" splitlines will add a lot of code, because it's currently implemented as something like str.split('\r').  But since splitlines has its very own config variable, I guess it makes sense to actually provide a full and proper implementation of it.

**Comment by dpgeorge** [comment]
Fixed by cc80c4dd59d0bdf130277fdb0705c871e1b1afa2.
